test/teiserver/game/balancer_server_test.exs

```diff
@@ -36,7 +83,147 @@ defmodule Teiserver.Game.BalancerServerTest do
     assert result == 4
   end
 
+  test "load some players in and run a balance pass or two" do
+    team_count = 2
+
+    arbitrary_rank_for_all_test_users = 10
+
+    players =
+      make_users_with_ranks([10, 20, 30, 40])
+
+    dbg(players)
+
+    players2 =
+      players
+      |> Enum.map(fn x -> Map.put(x, :userid, x.id) end)
+
+    dbg(players2)
+
+    {:ok, start_link_results} = BalancerServer.start_link(data: %{lobby_id: 1})
+
+    starting_state = GenServer.call(start_link_results, :report_state, 5000)
+    dbg(starting_state)
+
+    first_balance_pass =
+      GenServer.call(
+        start_link_results,
+        {:make_balance, team_count, [], players},
+        5000
+      )
+
+    state2 = GenServer.call(start_link_results, :report_state, 5000)
+    dbg(state2)
+
+    second_balance_pass_hash_reuse =
+      GenServer.call(
+        start_link_results,
+        {:make_balance, team_count, [], players},
+        5000
+      )
+
+    state3 = GenServer.call(start_link_results, :report_state, 5000)
+    dbg(state3)
+
+    assert second_balance_pass_hash_reuse.hash == first_balance_pass.hash
```

Comment:
Still debugging why this fails when I passed in the same state and player list. :/

Reply:
This is because the hash is computed with, amongst other things, the following options:
```elixir
          last_balance_hash_cache_hit: state.last_balance_hash_cache_hit,
          last_balance_hash_cache_miss: state.last_balance_hash_cache_miss,
          last_balance_hash: state.last_balance_hash,
          last_balance_result: state.last_balance_result
```
that is incorrect. The only thing that should be included in the hash are the input parameters for making balance, because that's basically what memoization is. If you include other stuff for the hash, it loses its purpose.

Reply:
Thanks for the tip, my mistake. Reverting to the original `opts` that were passed in to make_balance fixed the issue. (still making sure I've dotted my i's and crossed my t's before I mark this PR as ready for final review)